model/textparse/promparse.go

```diff
@@ -229,12 +232,23 @@ func (p *PromParser) Labels(l *labels.Labels) {
 
 	p.builder.Reset()
 	metricName := unreplace(s[p.offsets[0]-p.start : p.offsets[1]-p.start])
-	p.builder.Add(labels.MetricName, metricName)
+	if p.enableTypeAndUnitLabels {
+		p.builder.AddMetricIdentity(labels.MetricIdentity{
+			Name: metricName,
+			Type: p.mtype,
```

Comment:
I see tUnit in the promLexer but I noticed we're not including it here. I've also noticed that we don't handle tUnit when we call next() 🤔 

I don't remember how the PromTextFormat looks like, but if it has units, shouldn't we handle it as well?

Reply:
PromTextFormat does not have units and won't have in future. We should be able to present special labels though -- something to test!